src/main/java/net/neoforged/neoforge/network/registration/NetworkRegistry.java

```diff
@@ -518,7 +543,7 @@ public boolean onVanillaConnectionDetectedAtServer(ServerConfigurationPacketList
 
```

Comment:
We probably want to rename `onVanillaConnectionDetectedAtServer` to indicate that it also works for non-neoforge connection type. For example `onVanillaOrOtherConnectionDetectedAtServer` (assuming `ConnectionType` is renamed too).